### PR TITLE
[Core] Fix cgroup used memory calculation for Ray memory monitor

### LINKED ISSUE #42894: [Core] Update the cgroup2 memory accounting logic to exclude buffer/cached memory
### What happened + What you expected to happen

https://github.com/ray-project/ray/pull/42508

Right now, we are correctly accounting memory usage for cgroup1, but cgroup 2 may still contain cached/buffered memory. We should fix this by Ray 2.10

### Versions / Dependencies

master

### Reproduction script

n/a

### Issue Severity

None

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?

From oom killer's perspective, file page caches are reclaimable and can be used when kernel needs memory (the memory can be reclaimed by writing the data back to the original file) so we should exclude those when calculating cgroup used memory.

Before this PR, we only exclude part of the file page caches that's inactive. This PR excludes both active and inactive.

## Related issue number
Closes #42894

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/common/memory_monitor.cc [hunk 1]
```diff
@@ -258,24 +137,28 @@ int64_t MemoryMonitor::GetCGroupV2MemoryUsedBytes(const char *stat_path,
...
     RAY_LOG_EVERY_MS(WARNING, kLogIntervalMs)
-        << "Failed to parse cgroup v2 memory usage. memory.current "
-        << current_usage_bytes << " inactive " << inactive_file_bytes;
+        << "Failed to parse cgroup memory usage. memory usage " << current_usage_bytes
+        << " inactive file " << inactive_file_bytes << " active file "
+        << active_file_bytes;
     return kNull;
   }
-  return current_usage_bytes - inactive_file_bytes;
+  return current_usage_bytes - inactive_file_bytes - active_file_bytes;
```
Comment (jjyao) [inline src/ray/common/memory_monitor.cc:161]: This is the key.

**Review by WeichenXu123 — APPROVED** [review]

File: src/ray/common/memory_monitor.cc (hunk 1 above)
Comment (WeichenXu123) [inline src/ray/common/memory_monitor.cc:161]: Let's comment why we need to substract `active_file_bytes`:

we can put the link as a good reference: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-memory

and comment that 

`total_active_file + total_inactive_file == cache - shmem`

Comment (WeichenXu123) [reply src/ray/common/memory_monitor.cc:161]: and comment that cgroup v2 doesn't have `cache` and `shmem` fields, so we use `total_active_file, total_inactive_file` in formula to unify fomula across v1 / v2

**Review by WeichenXu123 — CHANGES_REQUESTED** [review]
You should update python side code too.

**Review by WeichenXu123 — APPROVED** [review]

**Review by WeichenXu123 — COMMENTED** [review]
Please refine comments before merging.
https://github.com/ray-project/ray/pull/43071/files#r1483902223

File: python/ray/_private/utils.py [hunk 2]
```diff
@@ -602,49 +602,35 @@ def get_num_cpus(
 
 
 # TODO(clarng): merge code with c++
-def get_cgroupv1_used_memory(memory_stat_filename, memory_usage_filename):
+def get_cgroup_used_memory(
+    memory_stat_filename, memory_usage_filename, inactive_file_key, active_file_key
```
Comment (rkooo567) [inline python/ray/_private/utils.py:606]: Can you add type annotation?

**Review by rkooo567 — APPROVED** [review]
Awesome. How do we plan to test it? Is it possible you run it with high IO workloads or are we just going to provide a wheel to users who have the problem?

**Pull request merged**